### PR TITLE
Remove remainder of astropy.tests.plugins.config plugin

### PR DESCRIPTION
**Pull request by astrofrog**
This is a follow-up to https://github.com/astropy/astropy/pull/7787 - it wasn't clear to me why we've retained the current simple function in ``config.py``? Was it for backward-compatibility?

In any case, we definitely should remove this plugin from ``conftest.py``.

### CODE REVIEW

**Comment by drdavella** [comment]
I'm guessing the reason for the test failure is it also needs to be removed here:
https://github.com/astropy/astropy/blob/4fe967fc5cda5c0eb8546810bbbf18235e4a7e1b/astropy/tests/runner.py#L415-L424

**Comment by pllim** [comment]
AFAIK removing it wasn't in the scope back then. If no one is using or needs this, I am not opposed to this PR.
```python
def pytest_configure(config):
    treat_deprecations_as_exceptions()
```

**Comment by astrofrog** [comment]
Well the only code that's still in that plugin has nothing to do with the plugin itself (we already enable warnings as deprecations inside ``conftest.py``, hence why I figured we can remove the rest.

**Comment by astrofrog** [comment]
So I think I'm confused about the purpose of this plugin. It looks like even though there's no code in it, its presence is still enough to trigger special behavior in the test runner:

```
        # If we are running the astropy tests with the astropy plugins handle
        # the config stuff, otherwise ignore it.
        if 'astropy.tests.plugins.config' not in plugins:
            return pytest.main(args=args, plugins=plugins)

        # override the config locations to not make a new directory nor use
        # existing cache or config
        astropy_config = tempfile.mkdtemp('astropy_config')
        astropy_cache = tempfile.mkdtemp('astropy_cache')

        # Have to use nested with statements for cross-Python support
        # Note, using these context managers here is superfluous if the
        # config_dir or cache_dir options to py.test are in use, but it's
        # also harmless to nest the contexts
        with set_temp_config(astropy_config, delete=True):
            with set_temp_cache(astropy_cache, delete=True):
                return pytest.main(args=args, plugins=plugins)
```

is that intentional? especially since this appears to be enabled by default... I think we should probably be using a command-line flag to modify the behavior?

**Comment by pllim** [comment]
Huh... I thought we already removed these options... @drdavella ?

```python
        astropy_config = tempfile.mkdtemp('astropy_config')
        astropy_cache = tempfile.mkdtemp('astropy_cache')
```

**Comment by drdavella** [comment]
The logic in the test runner appears to be completely independent of the plugin. In fact, it looks like it's overriding any options that might have been set by the plugin.

I guess the original purpose of the plugin was to allow these locations to be configured. I thought we determined that no one actually used this or cared about it, so it got removed in #7787. However, it looks like the test runner overrides these things anyway. But then it also uses the plugin anyway? It's very confusing.

Basically I think the only change that's required here for now is to remove the reference to the config plugin from this line: https://github.com/astropy/astropy/blob/4fe967fc5cda5c0eb8546810bbbf18235e4a7e1b/astropy/tests/runner.py#L415

**Comment by astrofrog** [comment]
Ok I understand better now - I've updated this PR.

**Comment by bsipocz** [comment]
Just a side note: before releasing 3.2, we should at least remind the affiliated packages once more that they should expect breakage unless they update their infrastructure. (And also make sure all coordinated packages are working).

**Comment by pllim** [comment]
Yeahhh, let's do this!
